### PR TITLE
add user-agent to Azure API calls, and support MSI/CLI cred for inventory

### PR DESCRIPTION
**Pull request by yuwzho**
##### SUMMARY

azure_rm.py 

##### ISSUE TYPE

 - Bugfix Pull Request

##### ANSIBLE VERSION

```
2.5.2
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: contrib/inventory/azure_rm.py [hunk 1]
```diff
@@ -299,9 +316,9 @@ def __init__(self, args):
         self.log("setting subscription_id")
         self.subscription_id = self.credentials['subscription_id']
 
-        if self.credentials.get('client_id') is not None and \
-           self.credentials.get('secret') is not None and \
-           self.credentials.get('tenant') is not None:
+        if self.credentials.get('credentials'):
+            self.azure_credentials = self.credentials.get('credentials')
```
Comment (zikalino) [inline contrib/inventory/azure_rm.py:320]: i think i would do:

self.azure_credentials = self.credentials.get('credentials')

if not self.azure_credentials:
    ....

**Comment by zikalino** [comment]
I think the fix looks good to me.
I am just concerned now whether we can just have some tests? Or only manual?

File: contrib/inventory/azure_rm.py (hunk 1 above)
Comment (yuwzho) [reply contrib/inventory/azure_rm.py:320]: It has 4 conditional check here, I would like to choose the shortest one. 😃

**Comment by yuwzho** [comment]
@zikalino Yes, here I test it manually,
Our inventory is lack of both integration and unit test.
I think we can start to setup an UT framework for our modules.

**Pull request merged**